pkg/apis/core/validation/validation.go

```diff
@@ -5930,6 +5932,27 @@ func ValidatePodStatusUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions
 	// ephemeral containers are not allowed to have resources allocated
 	allErrs = append(allErrs, validateContainerStatusNoAllocatedResourcesStatus(newPod.Status.EphemeralContainerStatuses, fldPath.Child("ephemeralContainerStatuses"))...)
 
+	if opts.AllowImageVolumeWithDigest {
+		for _, containerStatus := range newPod.Status.ContainerStatuses {
+			for _, volumeMountStatus := range containerStatus.VolumeMounts {
+				validatedField := fldPath.Child("containerStatuses", "volumeMounts", "volumeStatus")
```

Comment:
The correct fieldpath would be something like `fldPath.Child("containerStatuses").Index(i).Child("volumeMounts").Index(j).Child("volumeStatus")`